### PR TITLE
[Merged by Bors] - chore: deprecate `Setoid.Rel`

### PR DESCRIPTION
**Pull request by astrainfinita**
---

- [x] depends on: #16254

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Setoid/Basic.lean [hunk 1]
```diff
@@ -39,47 +36,55 @@ attribute [trans] Setoid.trans
 variable {α : Type*} {β : Type*}
 
 /-- A version of `Setoid.r` that takes the equivalence relation as an explicit argument. -/
+@[deprecated (since := "2024-08-29")]
 def Setoid.Rel (r : Setoid α) : α → α → Prop :=
   @Setoid.r _ r
 
-instance Setoid.decidableRel (r : Setoid α) [h : DecidableRel r.r] : DecidableRel r.Rel :=
+instance Setoid.decidableRel (r : Setoid α) [h : DecidableRel r.r] : DecidableRel r :=
```
Comment (urkud) [inline Mathlib/Data/Setoid/Basic.lean:43]: I guess, we no longer need this instance.

**Review by YaelDillies — APPROVED** [review]
Clearly an improvement, thanks!

maintainer merge

File: Mathlib/GroupTheory/GroupAction/Basic.lean [hunk 2]
```diff
@@ -373,11 +373,11 @@ def orbitRel : Setoid α where
...
 
 @[to_additive]
-theorem orbitRel_apply {a b : α} : (orbitRel G α).Rel a b ↔ a ∈ orbit G b :=
+theorem orbitRel_apply {a b : α} : orbitRel G α a b ↔ a ∈ orbit G b :=
   Iff.rfl
 
 @[to_additive]
-lemma orbitRel_r_apply {a b : α} : (orbitRel G _).r a b ↔ a ∈ orbit G b :=
+lemma orbitRel_r_apply {a b : α} : orbitRel G _ a b ↔ a ∈ orbit G b :=
   Iff.rfl
```
Comment (eric-wieser) [inline Mathlib/GroupTheory/GroupAction/Basic.lean:381]: These lemmas are now identical, so one of them should be deprecated.

Can you add a comment about the chosen naming convention to CoeFun instance?

Comment (eric-wieser) [reply Mathlib/GroupTheory/GroupAction/Basic.lean]: To whatever defines the coercion to function

**Comment by urkud** [comment]
Please merge `master`.

File: Mathlib/Data/Quot.lean [hunk 3]
```diff
@@ -26,6 +26,8 @@ run_cmd Lean.Elab.Command.liftTermElabM do
   Lean.Meta.registerCoercion ``Setoid.r
     (some { numArgs := 2, coercee := 1, type := .coeFun })
 
+-- No need to reference `Setoid.r` in the names of lemmas,
+-- since it is already registered as a coercion.
```
Comment (urkud) [inline Mathlib/Data/Quot.lean:30]: Please turn this into a docstring.

**Comment by urkud** [comment]
Otherwise LGTM. Thanks!
bors d+

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**